org.eclipse.acute/src/org/eclipse/acute/dotnetrun/DotnetRunDelegate.java

```diff
@@ -0,0 +1,223 @@
+package org.eclipse.acute.dotnetrun;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.ProcessBuilder.Redirect;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.ICoreRunnable;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.variables.IStringVariableManager;
+import org.eclipse.core.variables.VariablesPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.model.ILaunchConfigurationDelegate;
+import org.eclipse.debug.ui.ILaunchShortcut;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.console.ConsolePlugin;
+import org.eclipse.ui.console.IConsole;
+import org.eclipse.ui.console.IConsoleConstants;
+import org.eclipse.ui.console.IConsoleManager;
+import org.eclipse.ui.console.IConsoleView;
+import org.eclipse.ui.console.MessageConsole;
+import org.eclipse.ui.console.MessageConsoleStream;
+
+public class DotnetRunDelegate implements ILaunchConfigurationDelegate, ILaunchShortcut {
+
+	private static final String CONSOLE_NAME = ".NET Console";
+
+	@Override
+	public void launch(ISelection selection, String mode) {
+
+		if (selection instanceof IStructuredSelection) {
+			Iterator<Object> selectionIterator = ((IStructuredSelection) selection).iterator();
+			while (selectionIterator.hasNext()) {
+				Object element = selectionIterator.next();
+				IResource resource = null;
+				if (element instanceof IResource) {
+					resource = (IResource) element;
+				} else if (element instanceof IAdaptable) {
+					resource = ((IAdaptable) element).getAdapter(IResource.class);
+				}
+
+				if (resource != null) {
```

Comment:
It would probably be better to use the process factories: https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Fguide%2Fdebug_launch_processfactories.htm . IIRC, those have options to automatically redirect to console and usually interact better with Launch Configurations.